webservice/src/main/java/com/thoughtworks/aceleradora/service/UserService.java

```diff
@@ -17,4 +17,8 @@ public UserService (UserRepository repository){
     public User save(User user){
         return repository.save(user);
     }
+
+    public User findByEmail(String email) {
+        return this.repository.findByEmail(email).stream().findFirst().get();
```

Comment:
Email não deveria ser único? Me assusta que a gente pegue uma lista e pegue o primeiro sem saber quem ele é. Se a gente tá pegando o primeiro, é pq temos a suposição que só existe um. Se só existe um, a gente tem que forçar isso a ser verdade.

Reply:
tu diz com validações?

Reply:
Com validações, enforçando unique a nível de banco de dados e mudando o método do repository pra retornar um `Optional<User>`.